### PR TITLE
Rr add exit code to deploy test update 01112016

### PR DESCRIPTION
**Pull request by rruster**

### CODE REVIEW

**Comment by jconstance-amplify** [comment]
Might be worthwhile to update the unit tests to test for the return value of the various methods. I think the blue_green tests already do that.

File: disco_aws_automation/disco_deploy.py [hunk 1]
```diff
@@ -808,6 +829,7 @@ def update(self, dry_run=False, deployment_strategy=None):
             self.update_ami(random.choice(amis), dry_run, deployment_strategy)
         else:
             logger.info("No new 'tested' AMIs found.")
+            raise RuntimeError("No new 'tested' AMIs found.")
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_deploy.py]: I'm not sure if this should be an error

File: disco_aws_automation/disco_deploy.py [hunk 2]
```diff
@@ -800,6 +820,7 @@ def test(self, dry_run=False, deployment_strategy=None):
             self.test_ami(random.choice(amis), dry_run, deployment_strategy)
         else:
             logger.info("No 'untested' AMIs found.")
+            raise RuntimeError("No 'untested' AMIs found.")
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_deploy.py]: Not sure if those should be an error

File: disco_aws_automation/disco_deploy.py [hunk 3]
```diff
@@ -552,7 +568,11 @@ def handle_blue_green_ami(self, ami, pipeline_dict=None, old_group=None,
         if uses_elb:
             # Destroy the testing ELB
             self._disco_elb.delete_elb(hostclass, testing=True)
-        return False
+        if not smoke_tests:
+            reason = "AMI smoke test failed."
+        else:
+            reason = "AMI integration test failed."
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_deploy.py]: Nice

**Pull request merged**